src/responses/page-elements.ts

```diff
@@ -0,0 +1,79 @@
+import { Aggregation } from '../models/aggregation';
+
+/**
+ * The structure of the response body `hits` object returned from the PPS endpoint.
+ */
+export interface SearchResponseHits {
+  total: number;
+  returned: number;
+  hits: Record<string, unknown>[];
+}
+
+/**
+ * Valid page element names recognized & returned by the PPS
+ */
+export type PageElementName =
+  | 'uploads'
+  | 'reviews'
+  | 'collections'
+  | 'lending'
+  | 'web_archives'
+  | 'forum_posts';
+
+/**
+ * A map containing one or more page elements returned by the PPS, keyed by the
+ * name of the element.
+ */
+export interface PageElementMap
+  extends Partial<Record<PageElementName, unknown>> {
+  uploads?: UploadsPageElement;
+  reviews?: ReviewsPageElement;
+  collections?: CollectionsPageElement;
+  lending?: LendingPageElement;
+  web_archives?: WebArchivesPageElement;
+  forum_posts?: ForumPostsPageElement;
+}
+
+export type UploadsPageElement = HitsAggregationsPageElement;
+export type ReviewsPageElement = HitsAggregationsPageElement;
+export type CollectionsPageElement = HitsAggregationsPageElement;
+export interface LendingPageElement {
+  loans: Record<string, unknown>[];
+  waitlist: Record<string, unknown>[];
+  loan_history: Record<string, unknown>[];
```

Comment:
loan_history can get long, can/should we query for these fields independently of each other?

Reply:
I'll have to check how (whether) these are being paged at the PPS end, and whether it's possible to separate them. Agreed that we don't want to load the entirety of one's loan history all at once!